### PR TITLE
🔒️(frontend) add defense in-depth for XSS vulnerabilities 

### PR DESCRIPTION
**Pull request by sylvinus**
Avoid using HTML strings whenever possible, always try to use the react render to string functions with JSX.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved HTML sanitization in signature rendering and more robust attachment HTML handling.
  * Tightened message unquote patterns to reduce overmatching and parsing errors.

* **Refactor**
  * Standardized HTML generation/rendering paths for images, attachments and plain text to improve consistency.

* **Chores**
  * Added a make target and npm script to run frontend tests and update snapshots; updated dev shell invocation.

* **Tests**
  * Expanded tests and snapshots for attachment parsing, HTML decoding, and edge cases.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -255,6 +255,10 @@ front-test: ## run the frontend tests
 	$(COMPOSE) run --rm frontend-tools npm run test -- $${args:-${1}}
 .PHONY: front-test
 
+front-test-update: ## run the frontend tests and update snapshots
+	$(COMPOSE) run --rm frontend-tools npm run test -- --update
+.PHONY: front-test-update
+
```
Comment (jbpenrath) [inline Makefile:261]: IMO no need to add this new command. We can use `make front-test -- --update`

File: src/frontend/src/features/utils/mail-helper.test.tsx [hunk 2]
```diff
@@ -1,6 +1,9 @@
+import { vi } from 'vitest';
 import MailHelper, { SUPPORTED_IMAP_DOMAINS, ATTACHMENT_SEPARATORS } from './mail-helper';
 import DetectionMap from '@/features/i18n/attachments-detection-map.json';
 
+vi.mock('./errors', () => ({ handle: vi.fn() }));
```
Comment (jbpenrath) [inline src/frontend/src/features/utils/mail-helper.test.tsx:5]: Useless nope ?

File: src/frontend/src/features/utils/mail-helper.test.tsx [hunk 3]
```diff
@@ -340,14 +343,7 @@ describe('MailHelper', () => {
       expect(result).toMatchInlineSnapshot(`
         "<h1>Hello, how are you today?</h1>
         ---------- Drive attachments ----------
```
Comment (jbpenrath) [inline src/frontend/src/features/utils/mail-helper.test.tsx:345]: We could add a test with a malicious string

File: src/frontend/src/features/utils/mail-helper.tsx [hunk 4]
```diff
@@ -167,15 +167,16 @@
...
-            + a.name
-            + '</a>\n'
-            + '</li>'
-            ).join('\n')
-        + `\n</ul>\n\n`;
+        + renderToStaticMarkup(
+            <ul>
+                {attachments.map((a) => (
+                    <li key={a.id}>
+                        <a className="drive-attachment" href={a.url} data-id={a.id} data-name={a.name} data-type={a.type} data-size={String(a.size)} data-created_at={a.created_at}>{a.name}</a>
```
[the remark this answers is not in the document]
Comment (sylvinus) [reply src/frontend/src/features/utils/mail-helper.tsx:185]: @copilot you are missing the use of renderToStaticMarkup which prevents what you describe

File: src/frontend/src/features/utils/mail-helper.test.tsx (hunk 2 above)
Comment (sylvinus) [reply src/frontend/src/features/utils/mail-helper.test.tsx:5]: there was a huge log

File: Makefile (hunk 1 above)
Comment (sylvinus) [reply Makefile:261]: It's more for discoverability, I didn't know about --update
